File: app/src/main/java/com/lostincontext/mainscreen/MainScreenPresenterModule.java
package com.lostincontext.mainscreen;

import com.lostincontext.awareness.AwarenessModule;
import com.lostincontext.data.rules.repo.RulesRepositoryModule;

import dagger.Module;
import dagger.Provides;


@Module(includes = {AwarenessModule.class})
public class MainScreenPresenterModule {

    private final MainScreenContract.View view;

    public MainScreenPresenterModule(MainScreenContract.View view) {
        this.view = view;
    }

    @Provides
    MainScreenContract.View provideMainScreenContractView() {
        return view;
    }

}

File: app/src/main/java/com/lostincontext/mainscreen/MainScreenPresenterModule.kt
package com.lostincontext.mainscreen

import com.lostincontext.awareness.AwarenessModule

import dagger.Module
import dagger.Provides


@Module(includes = arrayOf(AwarenessModule::class))
class MainScreenPresenterModule(private val view: MainScreenContract.View) {

    @Provides
    internal fun provideMainScreenContractView(): MainScreenContract.View {
        return view
    }

}


